test/test_mutable_binheap.jl

```diff
@@ -52,11 +52,12 @@ end
 
 @testset "MutableBinheap" begin
 
-    vs = [4, 1, 3, 2, 16, 9, 10, 14, 8, 7]
-    vs2 = collect(enumerate(vs))
     ordering = Base.Order.By(last)
 
     @testset "construct heap" begin
+        vs = [4, 1, 3, 2, 16, 9, 10, 14, 8, 7]
+        vs2 = collect(enumerate(vs))
```

Comment:
The new implementation was overwriting `vs`. So, the tests were failing because of updated `vs` (The value of `vs` was not as expected because it was updated during some other test). So, I updated it to use separate `vs` and `vs2` for each test.

Reply:
It is not OK for `MutableBinaryMinHeap(vs)` to mutate `vs`.
We should add a `copy` to the constructor if that is the case.

(we can consider later adding a `MutableBinaryMinHeap!` or something that does do so. but lets not worry about it in this PR)

Reply:
`list_values` related tests (i.e., [this one](https://github.com/JuliaCollections/DataStructures.jl/blob/master/test/test_mutable_binheap.jl#L96)) started failing after adding copy in the constructor. the reason for that is because `nodemap` is not updated as I mentioned [here](https://github.com/JuliaCollections/DataStructures.jl/pull/712#issuecomment-742761512). Those tests were not failing before because `vs` was being mutated.

I am trying to think if there is an efficient way to update `nodemap` after the `heapify!` call.

It might be that not being able to update `nodemap` efficiently (with O(n) implementation) is the reason they implemented it in O(nlogn) in the first place.